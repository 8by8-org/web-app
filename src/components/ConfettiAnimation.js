import React, { useState, useEffect } from "react";
import Confetti from "react-confetti";
import { delay } from "./../functions/UserData";

function ConfettiAnimation(props) {
  // gets the total height of the page content
  var page_height = document.body.offsetHeight
    ? document.body.offsetHeight
    : document.height;

  const [maxWidth, setMaxWidth] = useState(window.innerWidth - 20);
  const [maxHeight, setMaxHeight] = useState(page_height);
  const [showConfetti, setShowConfetti] = useState(true);

  const [confetti, setConfetti] = useState(
    <Confetti
      width={maxWidth}
      height={maxHeight}
      recycle={showConfetti}
      numberOfPieces={200}
    />
  );

  useEffect(() => {
    startConfettiTimer(props.time);
    window.addEventListener("resize", () => {
      setMaxWidth(window.innerWidth - 20);

      page_height = document.body.offsetHeight
        ? document.body.offsetHeight
        : document.height;
      setMaxHeight(page_height);
    });
  }, []);

  useEffect(() => {
    setConfetti(
      <Confetti
        width={maxWidth}
        height={maxHeight}
        recycle={showConfetti}
        numberOfPieces={200}
      />
    );
  }, [maxHeight, maxWidth, showConfetti]);

  async function startConfettiTimer(time) {
    await delay(time);
    setShowConfetti(false);
  }

  return confetti;
}

export default ConfettiAnimation;
