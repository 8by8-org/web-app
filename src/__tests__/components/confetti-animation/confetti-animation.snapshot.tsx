import ConfettiAnimation from '@/components/confetti-animation/confetti-animation';
import { cleanup, render } from '@testing-library/react';

describe('ConfettiAnimation', () => {
  afterEach(cleanup);
  it('renders the ConfettiAnimation unchanged.', () => {
    const { container } = render(<ConfettiAnimation time={8000}/>);
    expect(container).toMatchSnapshot();
  });
});