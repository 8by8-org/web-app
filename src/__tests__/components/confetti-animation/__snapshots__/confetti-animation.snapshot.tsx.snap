// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfettiAnimation renders the ConfettiAnimation unchanged. 1`] = `
<div>
  <canvas
    data-testid="confetti animation"
    height="0"
    style="z-index: 2; position: absolute; pointer-events: none; top: 0px; left: 0px; bottom: 0px; right: 0px;"
    width="0"
  />
</div>
`;
